compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/parser/BLangNodeTransformer.java

```diff
@@ -393,9 +392,14 @@ public BLangNode transform(ParenthesisedTypeDescriptorNode parenthesisedTypeDesc
     public BLangNode transform(TupleTypeDescriptorNode tupleTypeDescriptorNode) {
         // TODO: Fully implement after tuple-type-desc is completed.
         BLangTupleTypeNode tupleTypeNode = (BLangTupleTypeNode) TreeBuilder.createTupleTypeNode();
-        SeparatedNodeList<TypeDescriptorNode> types = tupleTypeDescriptorNode.memberTypeDesc();
+        SeparatedNodeList<Node> types = tupleTypeDescriptorNode.memberTypeDesc();
         for (int i = 0; i < types.size(); i++) {
-            tupleTypeNode.memberTypeNodes.add(createTypeNode(types.get(i)));
+            Node node = types.get(i);
+            if (node.kind() == SyntaxKind.REST_TYPE) {
+                tupleTypeNode.restParamType = createTypeNode(types.get(i));
```

Comment:
Here types.get(i) will give you a `RestDescriptorNode`.I think you'l have to get the `typeDescriptor` field of it, to pass to `createTypeNode()` method.

Reply:
Changed.